file.cc

```diff
@@ -51,179 +33,96 @@ bool tryPossibleZipName( std::string const & name, std::string & copyTo )
-#endif
-#else
-  struct stat buf;
-
-  // EOVERFLOW rationale: if the file is too large, it still does exist
-  return stat( filename, &buf ) == 0 || errno == EOVERFLOW;
-#endif
+  File::Class f( filename, "rb" );
+  QByteArray byteArray{ f.readall() };
+  data.reserve( byteArray.size() );
+  data = std::vector< char >( byteArray.cbegin(), byteArray.cend() );
 }
 
-void Class::open( char const * filename, char const * mode ) 
+void Class::open( char const * mode )
 {
   QFile::OpenMode openMode = QIODevice::Text;
+
   const char * pch = mode;
-  while( *pch )
-  {
-    switch( *pch )
-    {
-      case 'r': openMode |= QIODevice::ReadOnly;
-                break;
-      case 'w': openMode |= QIODevice::WriteOnly;
-                break;
-      case '+': openMode &= ~( QIODevice::ReadOnly | QIODevice::WriteOnly );
-                openMode |= QIODevice::ReadWrite;
-                break;
-      case 'a': openMode |= QIODevice::Append;
-                break;
-      case 'b': openMode &= ~QIODevice::Text;
-                break;
-      default:  break;
+  while ( *pch ) {
+    switch ( *pch ) {
+      case 'r':
+        openMode |= QIODevice::ReadOnly;
+        break;
+      case 'w':
+        openMode |= QIODevice::WriteOnly;
+        break;
+      case '+':
+        openMode &= ~( QIODevice::ReadOnly | QIODevice::WriteOnly );
+        openMode |= QIODevice::ReadWrite;
+        break;
+      case 'a':
+        openMode |= QIODevice::Append;
+        break;
+      case 'b':
+        openMode &= ~QIODevice::Text;
+        break;
+      default:
+        break;
     }
     ++pch;
   }
 
-  f.setFileName( filename );
-
   if ( !f.open( openMode ) )
-    throw exCantOpen( std::string( filename ) + ": " + f.errorString().toUtf8().data() );
+    throw exCantOpen( f.fileName().toStdString() + ": " + f.errorString().toUtf8().data() );
 }
 
-Class::Class( char const * filename, char const * mode ) :
-  writeBuffer( 0 )
+Class::Class( std::string_view filename, char const * mode )
 {
-  open( filename, mode );
+  f.setFileName( QString::fromUtf8( filename.data(), filename.size() ) );
+  open( mode );
 }
 
-Class::Class( std::string const & filename, char const * mode )
-  : writeBuffer( 0 )
+void Class::read( void * buf, qint64 size )
 {
-  open( filename.c_str(), mode );
-}
-
-void Class::read( void * buf, qint64 size ) 
-{
-  if ( !size )
-    return;
-
-  if ( writeBuffer )
-    flushWriteBuffer();
-
-  qint64 result = f.read( reinterpret_cast<char *>( buf ), size );
+  qint64 result = f.read( static_cast< char * >( buf ), size );
 
   if ( result != size )
     throw exReadError();
 }
 
-size_t Class::readRecords( void * buf, qint64 size, size_t count ) 
+size_t Class::readRecords( void * buf, qint64 size, qint64 count )
 {
-  if ( writeBuffer )
-    flushWriteBuffer();
-
-  qint64 result = f.read( reinterpret_cast<char *>( buf ), size * count );
+  qint64 result = f.read( static_cast< char * >( buf ), size * count );
```

Comment:
static_cast vs interpret_cast

Reply:
For void* -> char*, I think different kinds of casts have no practically difference, but using reinterpret_cast will pop up a warning :sweat_smile: 

![image](https://user-images.githubusercontent.com/20123683/232191018-f178a58e-eb23-42cd-8653-289f6092296b.png)

Citing this https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Pro-type-reinterpretcast

`reinterpret_cast` is to cast even in situations that doesn't make sense (like image->string). The `reinterpret_cast` is harmful.

Reply:
They are the same in this simple case. https://godbolt.org/z/aPYGovKfd
![image](https://user-images.githubusercontent.com/20123683/232193328-a6239a78-05f2-4071-8041-fc887baee3df.png)